lib/matplotlib/legend.py

```diff
@@ -550,6 +550,9 @@ def __init__(self, parent, handles, labels,
                         break
                     except AttributeError:
                         pass
+        elif isinstance(labelcolor, str) and labelcolor == 'none':
```

Comment:
I guess I'd have thought these were redundant....

Reply:
If a numpy array come is than `elif obj == "none":` gets us the "ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()" issue.

I thought about merging this with the conditional above, but was trying to avoid causing a rebase for @Carloscerq in #20084.

Reply:
Don’t worry about it. I think I can handle it over the weekend, and, if it’s the case, I can remake the #20084 .